x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elasticsearch/MultilingualE5SmallInternalServiceSettings.java

```diff
@@ -8,42 +8,107 @@
 package org.elasticsearch.xpack.inference.services.elasticsearch;
 
 import org.elasticsearch.TransportVersion;
+import org.elasticsearch.TransportVersions;
 import org.elasticsearch.common.ValidationException;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.inference.ModelConfigurations;
+import org.elasticsearch.inference.SimilarityMeasure;
 import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.inference.services.ServiceFields;
 import org.elasticsearch.xpack.inference.services.ServiceUtils;
 import org.elasticsearch.xpack.inference.services.settings.InternalServiceSettings;
 
 import java.io.IOException;
 import java.util.Arrays;
 import java.util.Map;
+import java.util.Objects;
 
+import static org.elasticsearch.TransportVersions.ML_INFERENCE_L2_NORM_SIMILARITY_ADDED;
 import static org.elasticsearch.TransportVersions.ML_TEXT_EMBEDDING_INFERENCE_SERVICE_ADDED;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractSimilarity;
 
 public class MultilingualE5SmallInternalServiceSettings extends ElasticsearchInternalServiceSettings {
 
     public static final String NAME = "multilingual_e5_small_service_settings";
 
-    public MultilingualE5SmallInternalServiceSettings(int numAllocations, int numThreads, String modelId) {
+    static final int DIMENSIONS = 384;
+    static final SimilarityMeasure DEFAULT_SIMILARITY = SimilarityMeasure.COSINE;
+
+    private final SimilarityMeasure similarity;
+    private final int dimensions;
+
+    public MultilingualE5SmallInternalServiceSettings(
+        int numAllocations,
+        int numThreads,
+        String modelId,
+        SimilarityMeasure similarityMeasure,
+        int dimensions
+    ) {
         super(numAllocations, numThreads, modelId);
+
+        this.similarity = similarityMeasure;
+        this.dimensions = dimensions;
     }
 
     public MultilingualE5SmallInternalServiceSettings(StreamInput in) throws IOException {
         super(in.readVInt(), in.readVInt(), in.readString());
+
+        if (in.getTransportVersion().onOrAfter(TransportVersions.ML_INFERENCE_L2_NORM_SIMILARITY_ADDED)) {
+            similarity = in.readEnum(SimilarityMeasure.class);
+            dimensions = in.readVInt();
+        } else {
+            similarity = DEFAULT_SIMILARITY;
```

Comment:
In the other services we've allowed these to be `null`. Seems like we want these to be something and the dimensions were hardcoded before to `384`. In a future PR we can do the same thing that the other external services do which is make an inference request to determine the dimensions dynamically.

Reply:
Is there any point serialising or persisting these fields when they are immutable. dimensions cannot change and cosine similarity should be used for this model

Reply:
Yeah probably not, I'll remove this and just force them to be `384` and `cosine`.